signer/src/storage/postgres.rs

```diff
@@ -361,10 +396,74 @@ impl super::DbRead for PgStore {
 
     async fn get_pending_withdraw_requests(
         &self,
-        _chain_tip: &model::BitcoinBlockHash,
-        _context_window: usize,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: i32,
     ) -> Result<Vec<model::WithdrawRequest>, Self::Error> {
-        Ok(Vec::new()) // TODO(246): Write query + integration test
+        let stacks_chain_tip = self.get_stacks_chain_tip(chain_tip).await?;
+        sqlx::query_as!(
+            model::WithdrawRequest,
+            r#"
+            WITH RECURSIVE extended_context_window AS (
+                SELECT block_hash, parent_hash, confirms, created_at, 1 AS depth
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+                UNION ALL
+                SELECT
+                    parent.block_hash
+                  , parent.parent_hash
+                  , parent.confirms
+                  , parent.created_at
+                  , last.depth + 1
+                FROM sbtc_signer.bitcoin_blocks parent
+                JOIN extended_context_window last ON parent.block_hash = last.parent_hash
+                WHERE last.depth <= $3 + 1
+            ),
+            last_bitcoin_block AS (
+                SELECT
+                    block_hash
+                  , confirms
+                FROM extended_context_window
+                WHERE depth = $3 + 1
```

Comment:
I would go with an `order by` and `limit 1` clause here, mainly to be resilient to the start-up case where the `context_window` is greater than the number of bitcoin blocks in the database. I mean I'd replace the `where` clause with
```sql
ORDER BY depth DESC
LIMIT 1
```

Reply:
Nice, thanks for suggesting. I've updated now.